### PR TITLE
Document content type

### PR DESCRIPTION
**Pull request by dmytrokorotkyi**
Document content type has been added to file metadata.

### CODE REVIEW

File: Backend/AnimalRescue/AnimalRescue.Contracts.BusinessLogic/AnimalRescue.Contracts.BusinessLogic.csproj [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
   <ItemGroup>
     <ProjectReference Include="..\AnimalRescue.Contracts.Common\AnimalRescue.Contracts.Common.csproj" />
+    <ProjectReference Include="..\AnimalRescue.DataAccess.Mongodb\AnimalRescue.DataAccess.Mongodb.csproj" />
```
Comment (AlexeyDerepa) [inline Backend/AnimalRescue/AnimalRescue.Contracts.BusinessLogic/AnimalRescue.Contracts.BusinessLogic.csproj]: Contracts should not know about DAL

File: Backend/AnimalRescue/AnimalRescue.Contracts.BusinessLogic/Interfaces/IDocumentService.cs [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
-namespace AnimalRescue.Contracts.BusinessLogic.Interfaces
-{
-    public interface IDocumentService :
-        IBlOneItemQueryAsyncy<byte[]>,
-        IBlUploadFileAsync
-    {
-    }
-}
+﻿using AnimalRescue.Contracts.BusinessLogic.Interfaces.CRUD;
+using AnimalRescue.DataAccess.Mongodb.Models;
```
Comment (AlexeyDerepa) [inline Backend/AnimalRescue/AnimalRescue.Contracts.BusinessLogic/Interfaces/IDocumentService.cs]: Contracts should not know about DAL

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Bucket.cs [hunk 3]
```diff
@@ -1,67 +1,87 @@
...
+
+        private async Task<string> UploadAsync<T>(
+            Func<string, T, GridFSUploadOptions, CancellationToken, Task<ObjectId>> operation, 
+            T fileData, 
+            string fileName, 
+            string contentType)
+        {
+            Require.Objects.NotNull(fileData, nameof(fileData));
+            Require.Strings.NotNullOrWhiteSpace(fileName, nameof(fileName));
+
```
Comment (AlexeyDerepa) [inline Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Bucket.cs]: Please use this construction:

Require.Strings.NotNullOrWhiteSpace(contentType, nameof(contentType));

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Bucket.cs [hunk 4]
```diff
@@ -1,67 +1,87 @@
...
+            return objectId.ToString();
+        }
+
+        public async Task<BucketItem> GetFileBytesAsync(string fileId)
+        {
+            Require.Strings.NotNullOrWhiteSpace(fileId, nameof(fileId));
+
+            ObjectId objectId = ObjectId.Parse(fileId);
+            byte[] bytes = await gridFSBucket.DownloadAsBytesAsync(objectId);
```
Comment (AlexeyDerepa) [inline Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Bucket.cs]: Why not get data by one request to data base?

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Bucket.cs [hunk 5]
```diff
@@ -1,67 +1,87 @@
...
+            Require.Strings.NotNullOrWhiteSpace(fileName, nameof(fileName));
+
+            var gridFSUploadOptions = new GridFSUploadOptions
+            {
+                Metadata = new BsonDocument {
+                    { ContentTypeName, contentType }
+                }
+            };
+
+            ObjectId objectId = await operation(fileName, fileData, gridFSUploadOptions, default(CancellationToken));
```
Comment (psyfakerr) [inline Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Bucket.cs]: A cancellation token parameter is redundant

**Pull request merged**